core/src/main/java/com/radixdlt/genesis/olympia/converter/OlympiaStakesConverter.java

```diff
@@ -0,0 +1,166 @@
+    int numStakesInCurrentChunk = 0;
+
+    for (var i = 0; i < olympiaStakesGrouped.size(); i++) {
+      final var entry = olympiaStakesGrouped.get(i);
+      final var validatorIndex = entry.getKey();
+      final var currStakes = entry.getValue();
+      final var validator = validators.get(validatorIndex);
+
+      if (validator.totalStakeUnits().isZero()) {
+        // Shouldn't really happen, but better not to divide by 0 later on :)
+        // We can just skip the validators without any stake
+        continue;
+      }
+
+      final ECDSASecp256k1PublicKey validatorPublicKey;
+      try {
+        validatorPublicKey =
+            ECDSASecp256k1PublicKey.fromBytes(validator.publicKeyBytes().asBytes());
+      } catch (PublicKeyException e) {
+        throw new OlympiaToBabylonGenesisConverterException(
+            "Olympia validator public key is invalid", e);
+      }
+
+      ImmutableList.Builder<GenesisStakeAllocation> allocationsBuilder = ImmutableList.builder();
+
+      for (var j = 0; j < currStakes.size(); j++) {
+        final var stake = currStakes.get(j);
+        // stake_xrd_value = total_xrd_staked * stake_unit_amount / total_stake_units
+        final var xrdAmountBigInt =
+            stake
+                .stakeUnitAmount()
+                .toBigInt()
+                .multiply(validator.totalStakedXrd().toBigInt())
+                .divide(validator.totalStakeUnits().toBigInt());
+        final var decimalXrdAmount = Decimal.unsafeFromRawBigIntRepr(xrdAmountBigInt);
+
+        final var isLast = i == olympiaStakesGrouped.size() - 1 && j == currStakes.size() - 1;
+        final var account = accounts.get(stake.accountIndex());
+        final var accountAddress =
+            Address.virtualAccountAddress(account.publicKeyBytes().asBytes());
+
+        final var accountIndex = accountsForCurrentChunk.insertIfMissingAndGetIndex(accountAddress);
+        allocationsBuilder.add(
+            new GenesisStakeAllocation(UInt32.fromNonNegativeInt(accountIndex), decimalXrdAmount));
+        numStakesInCurrentChunk += 1;
+        if (numStakesInCurrentChunk >= config.maxStakesPerChunk() || isLast) {
+          stakesInCurrentChunkBuilder.add(tuple(validatorPublicKey, allocationsBuilder.build()));
+          allocationsBuilder = ImmutableList.builder();
+
+          chunksBuilder.add(
+              new GenesisDataChunk.Stakes(
+                  accountsForCurrentChunk.build(), stakesInCurrentChunkBuilder.build()));
+          accountsForCurrentChunk = new UniqueListBuilder<>();
+          stakesInCurrentChunkBuilder = ImmutableList.builder();
+          numStakesInCurrentChunk = 0;
+        }
+      }
+      final var lastAllocationsForCurrentValidator = allocationsBuilder.build();
+      if (!lastAllocationsForCurrentValidator.isEmpty()) {
+        stakesInCurrentChunkBuilder.add(
+            tuple(validatorPublicKey, lastAllocationsForCurrentValidator));
+      }
+    }
+    return chunksBuilder.build();
```

Comment:
I may be missing something, but it looks like we never add the last `stakesInCurrentChunkBuilder` chunk to `chunksBuilder`?

EDIT: Ahh ignore me, that's handled by the `isLast` flag.